### PR TITLE
The plugin log name and checkpoint name support the logtail mode.

### PR DESCRIPTION
**Pull request by quzard**
This pull request introduces several new methods and configurations to better manage plugin-specific settings in the `AppConfig` class and related components. The changes focus on adding flexibility for log names, version tags, and checkpoint file names based on the `logtail_mode` flag.

### New Methods and Configurations:

* [`core/app_config/AppConfig.cpp`](diffhunk://#diff-7c2e04058fb2ce8c0c2fa83dfabc4bdc6335414d223a61389352b10ffdff02edR511-R534): Added methods `GetPluginLogName`, `GetVersionTag`, and `GetGoPluginCheckpoint` to return different values based on the `logtail_mode` flag.
* [`core/app_config/AppConfig.h`](diffhunk://#diff-11daa00fcde94fe16cf00cce492d8d7c913a87637a13a933216a3965694a5665R60-R62): Declared the new methods `GetPluginLogName`, `GetVersionTag`, and `GetGoPluginCheckpoint`.

### Plugin Configuration Updates:

* [`core/go_pipeline/LogtailPlugin.cpp`](diffhunk://#diff-206c398a3b9b0eba7bb671d8865886ba829b9a4d47fe437195aaa698b17cb735R68-R70): Updated the `LogtailPlugin` constructor to use the new methods for setting `LoongcollectorPluginLogName`, `LoongcollectorVersionTag`, and `LoongcollectorCheckPointFile` in the plugin configuration.
* [`pkg/config/global_config.go`](diffhunk://#diff-2e42261290dec53c859dc4726a5585d027388265f8bae2b8eadd79cdd60cac78R41-R46): Added new fields for `LoongcollectorPluginLogName`, `LoongcollectorVersionTag`, and `LoongcollectorCheckPointFile` in the `GlobalConfig` struct and initialized them in `newGlobalConfig()`. [[1]](diffhunk://#diff-2e42261290dec53c859dc4726a5585d027388265f8bae2b8eadd79cdd60cac78R41-R46) [[2]](diffhunk://#diff-2e42261290dec53c859dc4726a5585d027388265f8bae2b8eadd79cdd60cac78R76-R78)

### Log and Checkpoint Management:

* [`pkg/logger/logger.go`](diffhunk://#diff-c1dc2eb1e149a1dc7ecd5cb474e167fb456eb7ad0ea0256cf62fba693a89af79L45-R45): Modified log file patterns to use the new plugin log name and updated the `generateDefaultConfig` function accordingly. [[1]](diffhunk://#diff-c1dc2eb1e149a1dc7ecd5cb474e167fb456eb7ad0ea0256cf62fba693a89af79L45-R45) [[2]](diffhunk://#diff-c1dc2eb1e149a1dc7ecd5cb474e167fb456eb7ad0ea0256cf62fba693a89af79L57-R57) [[3]](diffhunk://#diff-c1dc2eb1e149a1dc7ecd5cb474e167fb456eb7ad0ea0256cf62fba693a89af79L328-R328)
* [`pluginmanager/checkpoint_manager.go`](diffhunk://#diff-0518a54ae1689482c21e60ebee200861228f6f7a5f2b10b0b91b012142218bf2L33-R33): Updated the checkpoint manager to use the new checkpoint file name from the global configuration if not provided via command-line flag. [[1]](diffhunk://#diff-0518a54ae1689482c21e60ebee200861228f6f7a5f2b10b0b91b012142218bf2L33-R33) [[2]](diffhunk://#diff-0518a54ae1689482c21e60ebee200861228f6f7a5f2b10b0b91b012142218bf2R92-R96)

### Miscellaneous:

* [`core/pipeline/Pipeline.cpp`](diffhunk://#diff-e24971873d9cd824efa7a5a06001231859f350d7156636609397bd590822d2eeR23): Included `app_config/AppConfig.h` and updated log error messages to use the new `GetPluginLogName` method. [[1]](diffhunk://#diff-e24971873d9cd824efa7a5a06001231859f350d7156636609397bd590822d2eeR23) [[2]](diffhunk://#diff-e24971873d9cd824efa7a5a06001231859f350d7156636609397bd590822d2eeL501-R502) [[3]](diffhunk://#diff-e24971873d9cd824efa7a5a06001231859f350d7156636609397bd590822d2eeL520-R521)
* [`pluginmanager/plugin_manager.go`](diffhunk://#diff-e76a2d5dcc62862183887eb80c0bbc0619fcccb49ba9a406d4ec0f61f530ba11L60-R60): Updated JSON configuration templates to use the new version tag from the global configuration. [[1]](diffhunk://#diff-e76a2d5dcc62862183887eb80c0bbc0619fcccb49ba9a406d4ec0f61f530ba11L60-R60) [[2]](diffhunk://#diff-e76a2d5dcc62862183887eb80c0bbc0619fcccb49ba9a406d4ec0f61f530ba11L80-R80)
* [`pkg/logger/logger_test.go`](diffhunk://#diff-1b8d9e7644b6bb4b1cbd02d902261663a31e2f4792231472ee673265132a6913L52-R56): Adjusted test cleanup and log reading functions to use the new plugin log name.

### CODE REVIEW

File: core/application/Application.cpp [hunk 1]
```diff
@@ -174,9 +174,9 @@ void Application::Init() {
     appInfoJson["UUID"] = Json::Value(Application::GetInstance()->GetUUID());
     appInfoJson["instance_id"] = Json::Value(Application::GetInstance()->GetInstanceId());
 #ifdef __ENTERPRISE__
```
Comment (yyuuttaaoo) [inline core/application/Application.cpp:176]: 这里有希望把hostid打印出来吗？

**Review by yyuuttaaoo — APPROVED** [review]

File: core/app_config/AppConfig.cpp [hunk 2]
```diff
@@ -1646,7 +1686,7 @@ void AppConfig::SetLoongcollectorConfDir(const std::string& dirPath) {
     //     = AbsolutePath(STRING_FLAG(ilogtail_local_yaml_config_dir), mLogtailSysConfDir) + PATH_SEPARATOR;
```
Comment (messixukejia) [inline core/app_config/AppConfig.cpp:1686]: 这些代码如果彻底不用了就删掉吧

File: core/application/Application.cpp [hunk 3]
```diff
@@ -32,8 +32,8 @@
 #include "common/version.h"
 #include "config/ConfigDiff.h"
 #include "config/InstanceConfigManager.h"
-#include "config/watcher/PipelineConfigWatcher.h"
 #include "config/watcher/InstanceConfigWatcher.h"
+#include "config/watcher/PipelineConfigWatcher.h"
```
Comment (messixukejia) [inline core/application/Application.cpp:36]: 开源文档

File: core/app_config/AppConfig.cpp (hunk 2 above)
Comment (quzard) [reply core/app_config/AppConfig.cpp:1686]: 这周提个pr把这些代码删了

**Review by yyuuttaaoo — APPROVED** [review]

**Pull request merged**